### ISSUE TITLE
failure-threshold from config ignores 'style'

### ISSUE THREAD

**Issue by oliv3r** [hadolint/hadolint#716]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
The `failure-threshold` option should behave equally in a configuration file and from an argument.

### Actual behavior
`failure-threshold` seems to be only honored from the CLI, not the config file

### Steps to reproduce the behavior
* Create a `Dockerfile` with a style error in it; e.g. `LABEL Maintainer="a@b.com"` (Maintainer must be lower-caps)
* Create a `.hadolint.yaml` file `failure-threshold: "style"` in it.
* `docker run --rm --interactive hadolint/hadolint hadolint Dockerfile`
passes, due to style threshold not being failing
* `docker run --rm --interactive hadolint/hadolint hadolint --failure-threshold: "style" Dockerfile`
fails due to style error
*  `docker run --rm --interactive --volume "$(pwd):/workdir" -w "/workdir" hadolint/hadolint hadolint Dockerfile`
passes, as it ignores the failure threshold 'style' from the config file

Note, that other warnings work fine, e.g. 'upgrading' the style issue to an info issue, we do get a failure code.
```yaml
failure-threshold: "style"
override:
  info:
    - DL3048
```
Upgrading the threshold to "warning" then changes our exit behavior as expected.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v2.7.0-0-g3e13ec6
```

**Comment by m-ildefons** [comment]
Hi @oliv3r ,
thanks for the bug report. This is indeed a bug in the config file parsing logic. I'm working on a fix for this.
In the meantime, you can coerce Hadolint into your desired behavior by using the commandline options.

**Comment by oliv3r** [comment]
@m-ildefons yeah; I think the bug-report showed that kind of ;) thanks for the tip anyway, much appreciated. As the linter is part of a template, and the local config file should be leading; that's not really possible; but the the wait for a fixed release is not a problem.
